base/mmap.jl

```diff
@@ -57,9 +57,9 @@ grow!(::Anonymous,o::Integer,l::Integer) = return
 function grow!(io::IO, offset::Integer, len::Integer)
     pos = position(io)
     filelen = filesize(io)
-    if filelen < offset + len
```

Comment:
I'm not sure I understand how it's a race condition.  I do understand that this also changes the behavior in the way you said, and I had thought maybe that was a desirable behavior.  Now I see that this would cause problems if one wanted to mmap only part of a file and keep the rest intact.  I will update the pull request with your suggestions, thanks!